ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/numberinput/v1/numberinput/clientlibs/site/js/numberinputwidget.js

```diff
@@ -0,0 +1,408 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+class NumericInputWidget {
+    #widget=null
+    #model=null // passed by reference
+    #options=null
+    #defaultOptions={
+        value : null,
+        curValue: null,
+        pos: 0,
+        lengthLimitVisible: true,
+        zero:"0",
+        decimal:".",
+        minus:"-"
+    }
+    //TODO: to support writing in different locales \d should be replaced by [0-9] for different locales
+    #matchArray ={
+        "integer":"^[+-]?{digits}*$",
+        "decimal":"^[+-]?{digits}{leading}({decimal}{digits}{fraction})?$",
+        "float":"^[+-]?{digits}*({decimal}{digits}*)?$"
+    }
+    #regex=null
+    #processValue=null
+    #engRegex=null
+    #writtenInLocale=false
+    #previousCompositionVal=""
+
+    #toLatinForm    (halfOrFullWidthStr) {
+        // refer http://www.fileformat.info/info/unicode/block/halfwidth_and_fullwidth_forms/utf8test.htm
+        return halfOrFullWidthStr.replace(
+            /[\uff00-\uffef]/g,
+            function(ch) { return String.fromCharCode(ch.charCodeAt(0) - 0xfee0); }
+        );
+    }
+
+    constructor(widget, model) {
+        // initialize the widget and model
+        this.#widget = widget;
+        this.#model = model;
+        // initialize options for backward compatibility
+        this.#options = Object.assign({}, this.#defaultOptions, this.#model._jsonModel);
+        let matchStr =  this.#matchArray[this.#options.dataType];
+        if(matchStr) {
+            let ld = this.#options.leadDigits,
+                fd = this.#options.fracDigits,
+                ldstr = ld && ld !== -1 ? "{0,"+ld+"}"
+                    : "*",
+                fdstr = fd && fd !== -1 ? "{0,"+fd+"}"
+                    : "*",
+                matchStr =  matchStr.replace("{leading}",ldstr)
+                    .replace("{fraction}",fdstr),
+                localeStr = matchStr.replace(/{digits}/g,this.#getDigits())
+                    .replace("{decimal}",this.#escape(this.#options.decimal)),
+                engStr = matchStr.replace(/{digits}/g,"[0-9]")
+                    .replace("{decimal}","\\.")
+            this.#processValue = !(this.#getDigits() === "[0123456789]" && this.#options.decimal === ".")
+            this.#regex = new RegExp(localeStr, "g");
+            this.#engRegex = new RegExp(engStr, "g");
+        }
+        // change the input type to text for patterns
+        if (this.#options.editFormat || this.#options.displayFormat) {
```

Comment:
I'm assuming this widget itself is initialised when editFormat or displayFormat is there. So, is there a need of this check here again?